tests/ProxyManagerTest/Factory/LazyLoadingValueHolderFactoryTest.php

```diff
@@ -170,7 +167,6 @@ static function (ClassGenerator $targetClass) use ($proxyClassName) : bool {
         /** @var LazyLoadingMock $proxy */
         $proxy = $factory->createProxy($className, $initializer);
 
-        self::assertInstanceOf($proxyClassName, $proxy);
```

Comment:
This should most likely be kept

Reply:
Why? It will be catch by the interface. The return method signature.

Reply:
`$proxyClassName` is a dynamic name, coming from generated code that isn't inspected by static analysis.

Reply:
So yes, we know the proxy has a `: self` return type, but no static analysis tooling knows about that ;-)